extension/oauth2clientauthextension/extension.go

```diff
@@ -82,13 +95,27 @@ func (o *ClientCredentialsAuthenticator) Shutdown(_ context.Context) error {
 	return nil
 }
 
+func (ewts errorWrappingTokenSource) Token() (*oauth2.Token, error) {
+	tok, err := ewts.ts.Token()
+	if err != nil {
+		var errs error
+		errs = multierr.Append(errs, fmt.Errorf("%w (endpoint %q)", ErrFailedToGetSecurityToken, ewts.config.TokenURL))
+		errs = multierr.Append(errs, err)
```

Comment:
NIT: You can do `errs = multierr.Combine(errs, fmt.Errorf("%w (endpoint %q)", ErrFailedToGetSecurityToken, ewts.config.TokenURL), err)` single line instead of two calls